tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py

```diff
@@ -22,152 +19,134 @@
-registry_recrusion_512 = os.path.join(registry, 'recursion512')
+registry_recursion_1 = os.path.join(registry, 'recursion1')
+registry_recursion_5 = os.path.join(registry, 'recursion5')
+registry_recursion_29 = os.path.join(registry, 'recursion29')
+
+test_regs = [os.path.join(key, registry_no_recursion),
+             os.path.join(key, registry_recursion_1),
+             os.path.join(key, registry_recursion_5),
+             os.path.join(key, registry_recursion_29)]
+
+rl_dict = {
+    registry_no_recursion: '0',
+    registry_recursion_1: '1',
+    registry_recursion_5: '5',
+    registry_recursion_29: '29'
+}
+
+conf_params = {'REGISTRY_0': os.path.join(key, registry_no_recursion),
+               'LEVEL_0': rl_dict[registry_no_recursion],
+
+               'REGISTRY_1': os.path.join(key, registry_recursion_1),
+               'LEVEL_1': rl_dict[registry_recursion_1],
+
+               'REGISTRY_2': os.path.join(key, registry_recursion_5),
+               'LEVEL_2': rl_dict[registry_recursion_5],
+
+               'REGISTRY_3': os.path.join(key, registry_recursion_29),
+               'LEVEL_3': rl_dict[registry_recursion_29]
+}
 
-registry_list = [registry_no_recursion, registry_recrusion_1, registry_recrusion_5, registry_recrusion_512]
-test_regs = [registry_no_recursion, registry_recrusion_1, registry_recrusion_5, registry_recrusion_512]
 test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
 
 # Configurations
 
 configurations_path = os.path.join(test_data_path, "wazuh_recursion_windows_registry.yaml")
 wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
 
-p, m = generate_params(apply_to_all=({'FIM_MODE' : 'scheduled'}))
+p, m = generate_params(modes=['scheduled'], extra_params=conf_params)
 configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
 
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    return request.param
+
+
 def extra_configuration_before_yield():
-    """Make sure to delete any existing key with the same name before performing the test"""
-    for reg_key in registry_list:
-        create_registry(registry_parser[key], reg_key, 0, win32con.KEY_WOW64_32KEY)
-
-def extra_configuration_after_yield():
-    """Make sure to delete the key after performing the test"""
-    for reg_key in registry_list:
-        try:
-            delete_registry(registry_parser[key], reg_key, 0, win32con.KEY_WOW64_32KEY)
-        except win32api.error:
-            pass
-# Functions
-
-def recursion_test(key, registry, subkey, recursion_level, timeout=1, edge_limit=2, ignored_levels=1, is_scheduled=False):
+    for reg, rl in rl_dict.items():
+        path = str(reg)
+        for n in range(int(rl)):
+            path = os.path.join(path, '' + str(n + 1))
+        create_registry(registry_parser[key], path, KEY_WOW64_64KEY)
+
+# Tests
+
+@pytest.mark.parametrize('root_key, registry, arch, edge_limit, ignored_levels', [
```

Comment:
Please, add two blank lines before a function declaration.
```suggestion

@pytest.mark.parametrize('root_key, registry, arch, edge_limit, ignored_levels', [
```